crates/iceberg-catalog/src/service/storage/mod.rs

```diff
@@ -132,24 +153,31 @@ impl StorageProfile {
     ///
     /// # Errors
     /// Fails if the underlying storage profile's file IO creation fails.
-    pub fn file_io(
-        &self,
-        secret: Option<&StorageCredential>,
-    ) -> Result<iceberg::io::FileIO, FileIoError> {
+    pub fn file_io(&self, secret: Option<&StorageCredential>) -> Result<FileIO, FileIoError> {
         match self {
-            StorageProfile::S3(profile) => profile.file_io(
-                secret
-                    .map(|s| s.try_to_s3())
-                    .transpose()?
-                    .map(Into::into)
-                    .as_ref(),
-            ),
-            StorageProfile::Adls(prof) => prof.file_io(secret.map(|s| s.try_to_az()).transpose()?),
+            StorageProfile::S3(profile) => profile
+                .file_io(
+                    secret
+                        .map(|s| s.try_to_s3())
+                        .transpose()?
+                        .map(Into::into)
+                        .as_ref(),
+                )
+                .map(FileIO::FileIO),
+            StorageProfile::Adls(prof) => prof
+                .file_io(secret.map(|s| s.try_to_az()).transpose()?)
+                .map(FileIO::FileIO),
             #[cfg(test)]
-            StorageProfile::Test(_) => Ok(iceberg::io::FileIOBuilder::new("file").build()?),
+            StorageProfile::Test(_) => Ok(iceberg::io::FileIOBuilder::new("file")
+                .build()
+                .map(FileIO::FileIO)
+                .map_err(|e| FileIoError::FileIoCreationFailed(Box::new(e)))?),
             StorageProfile::Gcs(prof) => {
-                Ok(prof.file_io(secret.map(|s| s.try_into_gcs()).transpose()?)?)
+                Ok(prof.file_io(secret.map(|s| s.try_into_gcs()).transpose()?)?).map(FileIO::FileIO)
             }
+            StorageProfile::Hdfs(prof) => HdfsObjectStore::with_url(prof.url.as_str())
```

Comment:
change to with config constructor https://docs.rs/hdfs-native-object-store/0.13.0/hdfs_native_object_store/struct.HdfsObjectStore.html#method.with_config and pass in the string string hashmap from the profile

Reply:
url may not be needed because it may be read from hadoop config